terminus-ui/table/src/table.component.ts

```diff
@@ -229,31 +241,30 @@ export class TsTableComponent<T = any> extends CdkTable<T> implements OnInit, Af
    */
   public ngOnInit(): void {
     super.ngOnInit();
-
-    this.headerCellResizes.subscribe(v => {
-      const { instance, width } = v as TsHeaderCellResizeEvent;
-      this.updateColumnWidth(instance.columnDef.name, width);
-    });
-
-    this.headerCellHovers.subscribe(v => {
-      const { instance, isHovered } = v as TsHeaderCellResizeHoverEvent;
-      this.updateColumnHoverClass(instance.columnDef.name, isHovered);
-    });
+    this.initHeaderSubscriptions();
   }
 
 
   /**
    * Update the column width definitions when the rows change
    */
   public ngAfterViewChecked(): void {
+    this.viewChange.subscribe(v => {
+      this.initHeaderSubscriptions();
+    });
+
     this.rows.changes.subscribe(() => {
```

Comment:
is this `this.rows.changes` subset of that `this.viewChange`? So any time rows changed would also trigger `viewChange`?

Reply:
This took me forever to get working and I'm still not 100% sure why...

When columns are reordered I lose my event bindings, so I need to re-initialize them. With the inspector open you can visually see the rows DOM get updated, but for some reason the `rows.changes` doesn't fire (when using a trackBy or not). I ended up finding `viewChange` by digging through the CDK source code and it worked. I believe it's a CDR issue with `rows.changes` but I'm honestly not entirely sure. I've seen a couple places now that Material/CDK uses an internal Subject to alert others that the internals have changed rather than relying only on Angular cycles.